app/controllers/admin/commercials_controller.rb

```diff
@@ -57,7 +57,9 @@ def render_commercial
     def mass_upload
       errors = Commercial.read_file(params[:file]) if params[:file]
 
-      if errors.all? { |_k, v| v.blank? }
+      if !errors
```

Comment:
Are there any other conditions that could cause an error here?

Reply:
yes when user clicks on add commercial without selecting any file 'errors' in line 58^ is nil, it redirects him to error page: 

![](https://user-images.githubusercontent.com/13421233/34873339-bc93de6c-f7ba-11e7-997e-ea188b5d8dbd.png)

So, i thought there should be a error msg and redirect him back

Reply:
@bear454, is there any problem with this issue or PR? can you please review this :-) ?

Reply:
I think you misunderstood my question - I read your code as treating _any error_ as a result of a bad file upload; have you considered that _other_ conditions may cause errors ?

Reply:
oh, my bad, now I understand my mistake thanks :-)